src/Silverback.Integration/Diagnostics/IBrokerHeadersEnricher`1.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2020 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System.Diagnostics.CodeAnalysis;
+using Silverback.Messaging;
+
+namespace Silverback.Diagnostics
+{
+    /// <summary>
+    ///     Provides custom message headers.
+    /// </summary>
+    /// <typeparam name="TEndpoint">
+    ///     The type of the endpoint that this enricher can be used for.
+    /// </typeparam>
+    [SuppressMessage("ReSharper", "UnusedTypeParameter", Justification = "Used for DI")]
+    public interface IBrokerHeadersEnricher<TEndpoint> : IBrokerHeadersEnricher
+        where TEndpoint : Endpoint
```

Comment:
We should limit this to `ProducerEndpoint`, I don't see why you should ever enrich the consumed messages. (Or at least unless we decide to move other features to this pattern.)